src/Effects.js

```diff
@@ -1063,23 +1069,24 @@ module.exports = class Effects {
       draw: function () {
         p5.background('black');
         if (this.laser.length < 32) {
-          let laser = {
-            w: 200,
-            x: 200,
-            y: 1750,
-            z: 400,
-            color: lerpColorFromPalette(p5.frameCount / 16),
-          };
-          this.laser.push(laser);
+          const numLasersToDraw = this.getPreviewCustomizations().numLasersToDraw;
+          for (let i = 0; i < numLasersToDraw; i++) {
+            let laser = {
+              y: this.getPreviewCustomizations().getY(i),
+              z: 400,
+              color: this.getPreviewCustomizations().getColor(i)
+            };
+            this.laser.push(laser);
+          }
         }
         p5.push();
         p5.translate(200, 180);
-        for (const laser of this.laser) {
-          let y = 200 * p5.sin(p5.frameCount);
-          if (y < 0) {
-            y *= -1;
+        for (const [index, laser] of this.laser.entries()) {
+          let x = this.getPreviewCustomizations().getX(index);
+          if (x < 0) {
+            x *= -1;
           }
-          const angle = p5.atan2(laser.y, y);
+          const angle = p5.atan2(laser.y, x);
```

Comment:
According to p5 docs, second argument to `atan2` is x, so I renamed. If that doesn't make sense to ppl who remember trig better than I do, let me know :)

https://p5js.org/reference/#/p5/atan2

Reply:
Also, the w, x, and z parameters are unused, so I removed them.

Reply:
So much math!